bot/src/discord/schedule_helper.js

```diff
@@ -0,0 +1,370 @@
+const dbConnection = require('kotoba-node-common').database.connection;
+const WordScheduleModel = require('kotoba-node-common').models.createWordSchedulesModel(dbConnection);
+const showRandomWord = require('./show_random_word.js');
+
+const frequencyCheck = 30 * 60 * 1000;
+
+const statusConstants = {
+  running: 'running',
+  paused: 'paused',
+};
+
+// Returns the channel for this WOTD schedule, or undefined
+// if we cannot get the channel or cannot send messages in it.
+function getWOTDChannel(eris, serverId, channelId) {
+  const server = eris.guilds.get(serverId);
+
+  // If Kotoba is no longer in the server, return undefined.
+  if (!server) {
+    return undefined;
+  }
+
+  const channel = server.channels.get(channelId);
+
+  // If it's a guild channel, verify correct permissions.
+  if (channel.guild) {
+    // This checks that the channel is a TextChannel. Only they have this function.
+    if (!channel.permissionsOf) {
+      return undefined;
+    }
+
+    // Permissions that Kotoba has in this channel
+    const permissions = channel.permissionsOf(eris.user.id);
+
+    // Make sure we can send messages
+    if (!permissions.has('sendMessages')) {
+      return undefined;
+    }
+  }
+
+  // All good
+  return channel;
+}
+
+/**
+ * Starts the interval
+ * @param {Monochrome} monochrome
+ * @param {boolean} firstCall
+ */
+async function setTimer(monochrome, firstCall) {
+  let nextTime;
+  let now = new Date();
+  let offset = now.getTime() % frequencyCheck;
+  now -= offset; // We sync the time with the frequencyCheck value
```

Comment:
This converts `now` to a Number. I think that explains why the query needed the extra cast.

Reply:
Oh, I see. I'll replace it with:
```js
now = new Date(now - offset);
```